### PR TITLE
#193: Add push and build test-container CLI commands

### LINKED ISSUE #193: Add build and push test container CLI commands
Commands `build test container` and `push test container` are depending on the user of the test container.
Therefore ITDE has been changed to still provide the API for these commands but no CLI anymore.

As the commands are still required by the SLC this ticket requests to add an implementation here.

### PR DESCRIPTION
**Pull request by tkilias**
Fixes #193

### CODE REVIEW

File: test/test_build_test_container.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        utils.close_environments(self.test_environment)
+
+    def test_build_test_container(self):
+        parameter = exaslct_utils.get_mock_test_container_folder_parameter()
+        command = f"{self.test_environment.executable} build-test-container {parameter}"
+        self.test_environment.run_command(command, track_task_dependencies=True, use_flavor_path=False)
+        images = find_images_by_tag(self.docker_client,
+                                    lambda tag: tag.startswith(self.test_environment.repository_name))
+        self.assertTrue(len(images) > 0,
+                        f"Did not found images for repository "
```
Comment (ckunki) [inline test/test_build_test_container.py:33]:
```suggestion
                        f"Did not find images for repository "
```

File: test/test_build_test_container.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+import docker
+
+from exasol_script_languages_container_tool.lib.utils.docker_utils import find_images_by_tag
+
+import utils as exaslct_utils
+from exasol_integration_test_docker_environment.testing import utils
+
+
+class BuildTestContainerTest(unittest.TestCase):
```
Comment (ckunki) [inline test/test_build_test_container.py:11]: A general remark: I have the impression that the word "test" is used far too often.

At first glance there seem to be the following significantly different semantic domains:
1. The so-called "test container" in opposite to the "database container" with the test container designed to execute the tests verifying the behavior of the Exasol database inside the database container.
2. "test" code in opposite to "production code", with test code designed to verify the validity of the production features, avoid regressions and bugs.
3. "test" as the general purpose of SLC ITDE, i.e. environments created for the purpose of evaluating the Exasol Database inside a Docker Container (or maybe also in other scenarios), potentially in combination with other components that could be in other Docker Containers as well.

In my eyes this situation creates the problem that the names of many (really many) files, classs, attributes, methods, and variables contain the word "test" but for a reader it is not immediately clear which semantic domain this word is intended to link to, see the following examples:
- `class BuildTestContainerTest(unittest.TestCase):`
- `self.test_environment`
- `test/test_build_test_container.py` (3 times "test", 36 percent of all characters in the path)
- `def test_build_test_container(self):`

In summary I propose to use different words for each different semantic domain.

Candidates for the semantic domains listed above could be
1. test code execution, test orchestration, test runner, laboratory, testbench, qa, audit, isolation, QA Container, test logic container, Validation Container
2. "test"
3. evaluation, rating, environment, sandbox (probably used somewhere else already), docker-based, experiment, TestSuite, Product Evaluation Suite, Product Evaluation Docker Environment, QA Docker Environment, Fertigung, Prüf,

**Review by ckunki — CHANGES_REQUESTED** [review]

File: test/test_build_test_container.py (hunk 2 above)
Comment (tkilias) [reply test/test_build_test_container.py:11]: Please, create a ticket for it, in some cases I agree, in other cases, such as the test suites, they test frameworks used required the word test in specific places to recognize tests, they should run.

Comment (tkilias) [reply test/test_build_test_container.py:11]: see https://github.com/exasol/integration-test-docker-environment/issues/320

**Review by ckunki — APPROVED** [review]
